tools/js-tools/git-hooks/pre-push-hook.js

```diff
@@ -40,6 +40,54 @@ function checkChangelogFiles() {
 			)
 		);
 	} else {
+		try {
+			// Run the changelogger.
+			const autoChangelog = spawnSync( 'jetpack', [ 'changelog', 'add' ], {
+				stdio: 'inherit',
+			} );
+
+			// If the autochangelogger worked, commit the changelog files.
+			if ( autoChangelog.status === 0 ) {
+				const filesToCommit = [];
+				const changelogFiles = await spawnSync( 'git', [
+					'diff',
+					'--name-only',
+					'--diff-filter=A',
+					'--cached',
+				] )
+					.stdout.toString()
+					.trim()
+					.split( '\n' );
+
+				for ( const file of changelogFiles ) {
+					const match = file.match( /^projects\/([^/]+\/[^/]+)\/changelog\// );
+					if ( match ) {
+						filesToCommit.push( file );
+					}
+				}
+
+				if ( filesToCommit.length > 0 ) {
+					const commitFiles = await spawnSync( 'git', [
+						'commit',
+						...filesToCommit,
+						'-m',
+						'changelog',
+					] );
+					if ( commitFiles.status === 0 ) {
+						console.log(
+							'\r\n\r\n   _____ _____ _______   _____  _    _  _____ _    _            _____          _____ _   _ \r\n  / ____|_   _|__   __| |  __ \\| |  | |/ ____| |  | |     /\\   / ____|   /\\   |_   _| \\ | |\r\n | |  __  | |    | |    | |__) | |  | | (___ | |__| |    /  \\ | |  __   /  \\    | | |  \\| |\r\n | | |_ | | |    | |    |  ___/| |  | |\\___ \\|  __  |   / /\\ \\| | |_ | / /\\ \\   | | | . ` |\r\n | |__| |_| |_   | |    | |    | |__| |____) | |  | |  / ____ \\ |__| |/ ____ \\ _| |_| |\\  |\r\n  \\_____|_____|  |_|    |_|     \\____/|_____/|_|  |_| /_/    \\_\\_____/_/    \\_\\_____|_| \\_|\r\n                                                                                           \r\n                                                                                           \r\n\r\n'
```

Comment:
Strikes me as a bit excessive, but anyway, let's drop some of the excess whitespace
```suggestion
							'\n\n   _____ _____ _______   _____  _    _  _____ _    _            _____          _____ _   _\n  / ____|_   _|__   __| |  __ \\| |  | |/ ____| |  | |     /\\   / ____|   /\\   |_   _| \\ | |\n | |  __  | |    | |    | |__) | |  | | (___ | |__| |    /  \\ | |  __   /  \\    | | |  \\| |\n | | |_ | | |    | |    |  ___/| |  | |\\___ \\|  __  |   / /\\ \\| | |_ | / /\\ \\   | | | . ` |\n | |__| |_| |_   | |    | |    | |__| |____) | |  | |  / ____ \\ |__| |/ ____ \\ _| |_| |\\  |\n  \\_____|_____|  |_|    |_|     \\____/|_____/|_|  |_| /_/    \\_\\_____/_/    \\_\\_____|_| \\_|\n\n'
```

Or we could be even more excessive
```suggestion
							'\n\n\x1b[5m   _____ _____ _______   _____  _    _  _____ _    _            _____          _____ _   _\x1b[0m\n\x1b[5m  / ____|_   _|__   __| |  __ \\| |  | |/ ____| |  | |     /\\   / ____|   /\\   |_   _| \\ | |\x1b[0m\n\x1b[5m | |  __  | |    | |    | |__) | |  | | (___ | |__| |    /  \\ | |  __   /  \\    | | |  \\| |\x1b[0m\n\x1b[5m | | |_ | | |    | |    |  ___/| |  | |\\___ \\|  __  |   / /\\ \\| | |_ | / /\\ \\   | | | . ` |\x1b[0m\n\x1b[5m | |__| |_| |_   | |    | |    | |__| |____) | |  | |  / ____ \\ |__| |/ ____ \\ _| |_| |\\  |\x1b[0m\n\x1b[5m  \\_____|_____|  |_|    |_|     \\____/|_____/|_|  |_| /_/    \\_\\_____/_/    \\_\\_____|_| \\_|\x1b[0m\n\n'
```
(if you try that and it doesn't seem to do anything, lucky you)

Reply:
> (if you try that and it doesn't seem to do anything, lucky you)
Didn't seem to do anything :D 

The ASCII stuff is a compromise since we can't include the commit in the pre-push hook without wrapping everything. Figure better to be obnoxious than have people think the adding is failing.

Reply:
:D